runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -456,7 +477,36 @@ public static PackageAttributes forFileToStage(File source, String stagingPath)
       target.setName(uniqueName);
       target.setLocation(resourcePath);
 
-      return new AutoValue_PackageUtil_PackageAttributes(source, target, size, hash);
+      return new AutoValue_PackageUtil_PackageAttributes(source, null, target, size, hash);
+    }
+
+    public static PackageAttributes forBytesToStage(byte[] bytes, String targetName, String stagingPath)
+        {
+      long size;
+      String hash;
+      Hasher hasher = Hashing.md5().newHasher();
+      OutputStream hashStream = Funnels.asOutputStream(hasher);
+      try (CountingOutputStream countingOutputStream = new CountingOutputStream(hashStream)) {
+        ByteSource.wrap(bytes).copyTo(countingOutputStream);
+        countingOutputStream.flush();
+        size = countingOutputStream.getCount();
```

Comment:
Why is size not just the length of the bytes array?

In the above method it looks like we just used the same methodology for files and for directories to count their size instead of short circuiting and using the length of the file as the size.

Reply:
Done